File: backend/src/main/java/com/example/restapiagenceimmobiliere/dao/models/Annonce.java
package com.example.restapiagenceimmobiliere.dao.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.ZonedDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Annonce {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String titre;
    private String description;
    private double superficie;
    private int nombreDePieces;
    private String localisation;
    private double prix;
    
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(
        name = "annonce_photos",
        joinColumns = @JoinColumn(name = "annonce_id")
    )
    @Column(name = "photo_url")
    private List<String> photos;
    
    private String contact;
    private ZonedDateTime date_publication;
    
    @Enumerated(EnumType.STRING)
    private TypeBien typeBien;
    
    @Enumerated(EnumType.STRING)
    private Categorie categorie;
} 

File: frontend/src/app/shared/annonce.ts
import { Categorie } from "./categorie";
import { TypeBien } from "./type-bien";

export class Annonce {
    constructor(
        public id: number,
        public titre: string,
        public description: string,
        public superficie: number,
        public pieces: number,
        public localisation: string,
        public prix: number,
        public photos: string[],
        public contact: string,
        public typeBien: string,
        public categorie: string
    ) {}

    // Méthode pour convertir en objet à envoyer au backend
    toBackendObject(): any {
        return {
            id: this.id,
            titre: this.titre,
            description: this.description,
            superficie: this.superficie,
            nombre_de_pieces: this.pieces,
            localisation: this.localisation,
            prix: this.prix,
            photos: this.photos,
            contact: this.contact,
            typeBien: this.typeBien,
            categorie: this.categorie
        };
    }

    // Méthode statique pour créer un objet Annonce à partir d'une réponse backend
    static fromBackendObject(data: any): Annonce {
        return new Annonce(
            data.id,
            data.titre || '',
            data.description || '',
            data.superficie || 0,
            data.nombre_de_pieces || 0,
            data.localisation || '',
            data.prix || 0,
            data.photos || [],
            data.contact || '',
            data.typeBien || '',
            data.categorie || ''
        );
    }
}
